src/java/org/apache/cassandra/db/marshal/TupleType.java

```diff
@@ -196,48 +196,81 @@ public boolean isTuple()
         {
             int size = accessor.getInt(v, offset);
             offset += TypeSizes.INT_SIZE;
-            if (size > 0)
+            if (size >= 0)
```

Comment:
Because the change to '>' from CORE-92 causes empty strings / blobs to be interpreted as nulls which is incorrect and fails multiple unit tests.

The underlying problem that change was trying to fix was the fact that tuples can have nulls that are distinct from the null value for the specific type. Some types, e.g. ints, can store nulls directly -- AFAICS they make some sense in a clustering key, but in a tuple / user type nulls are handled by the tuple itself. The correct fix for this problem was thus to change the byte-comparable test for `TupleType` to only use tuple nulls.

Reply:
ok, thanks